src/python/bcc/__init__.py

```diff
@@ -62,13 +62,47 @@ def resolve_name(self, name):
             return -1
         return addr.value
 
+class PerfType:
+    # From perf_type_id in uapi/linux/perf_event.h
+    HARDWARE = 0
+    SOFTWARE = 1
```

Comment:
right now it's hw and sw only. Adding support for hw_cache and raw would be nice. If somebody can craft a kernel patch and test case, I can help upstreaming it.